src/content/docs/pl/basics/astro-components.mdx

```diff
@@ -0,0 +1,385 @@
+
+:::tip
+Możesz nawet używać TypeScript w swoim skrypcie komponentu!
+:::
+
+### Szablon komponentu
+
+Szablon komponentu znajduje się poniżej code fence i określa to, jaki HTML generuje Twój komponent.
+
+Jeżeli używasz w tym miejscu tylko znaczników HTML, Twój komponent wyrenderuje go na dowolnej stronie Astro, gdzie zostanie zaimportowany i użyty.
+
+Jednak [składnia szablonu komponentu Astro](/pl/basics/astro-syntax/) obsługuje również **wyrażenia JavaScript**, tagi [`<style>`](/pl/guides/styling/#styling-in-astro) i [`<script>`](/pl/guides/client-side-scripts/#using-script-in-astro) wspierane przez Astro, **importowane komponenty**, oraz [**specjalne dyrektywy Astro**](/pl/reference/directives-reference/). Dane i wartości zdefiniowane w skrypcie komponentu mogą być używane w szablonie komponentu do dynamicznego generowania HTMLa.
+
+```astro title="src/components/MyFavoritePokemon.astro"
+---
+// Twój skrypt komponentu tutaj!
+import Banner from '../components/Banner.astro';
+import ReactPokemonComponent from '../components/ReactPokemonComponent.jsx';
+const myFavoritePokemon = [/* ... */];
+const { title } = Astro.props;
+---
+<!-- Komentarze HTML są wspierane! -->
+{/* Składnia komentarza JavaScript również jest wspierana! */}
+
+<Banner />
+<h1>Witaj świecie!</h1>
+
+<!-- Użyj propsów i innych zmiennych z skryptu komponentu: -->
+<p>{title}</p>
+
+<!-- Dołącz komponenty frameworka UI za pomocą dyrektywy `client:` w celu przeprowadzenia hydracji: -->
+<ReactPokemonComponent client:visible />
+
+<!-- Mieszaj HTML z wyrażeniami JavaScript, podobnie jak w JSX: -->
+<ul>
+  {myFavoritePokemon.map((data) => <li>{data.name}</li>)}
+</ul>
+
+<!-- Użyj dyrektywy szablonu do budowania nazw klas z wielu ciągów lub nawet obiektów! -->
+<p class:list={["add", "dynamic", {classNames: true}]} />
+```
+
+## Projektowanie oparte na komponentach
+
+Komponenty zostały zaprojektowane w taki sposób, aby można było ich **reużywać w wielu miejscach** i **łączyć w większe jednostki**. Możesz używać komponentów wewnątrz innych komponentów, aby budować coraz bardziej zaawansowane elementy interfejsu użytkownika. Na przykład komponent `Button` może być użyty do stworzenia komponentu `ButtonGroup`:
+
+```astro title="src/components/ButtonGroup.astro"
+---
+import Button from './Button.astro';
+---
+<div>
+  <Button title="Przycisk 1" />
+  <Button title="Przycisk 2" />
+  <Button title="Przycisk 3" />
+</div>
+```
+
+
+## Propy w komponentach
+
+Komponent Astro może definiować i akceptować propy, które później stają się dostępne w szablonie komponentu i mogą zostać użyte w czasie renderowania HTMLa. Propy dostępne są w globalnej zmiennej `Astro.props` w twoim skrypcie przedmowy.
```

Comment:
```suggestion
Komponent Astro może definiować i akceptować propy, które później stają się dostępne w szablonie komponentu i mogą zostać użyte w czasie renderowania HTMLa. Propy dostępne są w globalnej zmiennej `Astro.props` wewnątrz skryptu frontmattera.
```
Since we agreed to stick with frontmatter